articles/architecture-scenarios/implementations/mobile-api/mobile-implementation-android.md

```diff
@@ -0,0 +1,1271 @@
+                        });
+                    }
+
+                    @Override
+                    public void onSuccess(@NonNull final Credentials credentials) {
+                        runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(LoginActivity.this, "Log In - Success", Toast.LENGTH_SHORT).show();
+                            }
+                        });
+                        startActivity(new Intent(LoginActivity.this, TimeSheetActivity.class));
+                    }
+                });
+    }
+}
+```
+
+### Store Credentials
+
+To store the credentials received after login, we’ll use the `CredentialsManager` from the Auth0.Android library and [SharedPreferences](https://developer.android.com/reference/android/content/SharedPreferences.html) for storage.
+
+Before initializing the `WebAuthProvider` in the `login()` method, we can create the `CredentialsManager`. Passing an `AuthenticationAPIClient` to the `CredentialsManager` enables it to refresh the access tokens if they are expired.
+
+```java
+private void login() {
+        Auth0 auth0 = new Auth0(getString(R.string.auth0_client_id), getString(R.string.auth0_domain));
+        auth0.setOIDCConformant(true);
+
+        AuthenticationAPIClient authAPIClient = new AuthenticationAPIClient(auth0);
+        SharedPreferencesStorage sharedPrefStorage = new SharedPreferencesStorage(this);
+        final CredentialsManager credentialsManager = new CredentialsManager(authAPIClient, sharedPrefStorage);
+
+        WebAuthProvider.init(auth0)
+        // ...
+    }
+```
+
+Now update the `login()` method so that credentials are stored via the `CredentialsManager` after a successful authentication.
+
+```java
+// ...
+@Override
+public void onSuccess(@NonNull final Credentials credentials) {
+    runOnUiThread(new Runnable() {
+        @Override
+        public void run() {
+            Toast.makeText(LoginActivity.this, "Log In - Success", Toast.LENGTH_SHORT).show();
+        }
+    });
+
+    credentialsManager.saveCredentials(credentials);
+    startActivity(new Intent(LoginActivity.this, TimeSheetActivity.class));
+// ...
+}
+```
+
+## 3. Get the User Profile
+
+### Create the User Model
+
+Create a simple User model that will be utilized by the `UserProfileManager` and `UserActivity`.
+
+```java
+package com.auth0.samples.models;
+
+public class User {
+    private String email;
+    private String name;
+    private String pictureURL;
+
+    public User(String gEmail, String gName, String gPictureURL) {
+        this.email = gEmail;
+        this.name = gName;
+        this.pictureURL = gPictureURL;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getPictureURL() {
+        return pictureURL;
+    }
+}
+```
+
+### Store User Profile
```

Comment:
store the user profile (make sure no links get broken, do this every time you change a header)